pkg/querier/queryrange/split_by_interval.go

```diff
@@ -122,7 +124,16 @@ func (h *splitByInterval) Process(
 				return nil, data.err
 			}
 
-			responses = append(responses, data.resp)
+			if casted, ok := data.resp.(*LokiResponse); ok {
+				if res == nil {
+					res = casted
+				}
+				if casted.Count() > 0 {
+					responses = append(responses, data.resp)
+				}
+			} else {
+				responses = append(responses, data.resp)
+			}
```

Comment:
if the correct thing to do here is to append the response even if we couldn't type assert it to `LokiResponse` then I think a comment would be useful, because it's not clear to me why we would want to do that